### PR TITLE
Added a condition to check whether an entity is an adult or not

### PR DESCRIPTION
**Pull request by DeltaRays**
**Target Minecraft Versions:** any 
**Requirements:** none 
**Related Issues:** none

### CODE REVIEW

File: src/main/java/ch/njol/skript/conditions/CondIsAdult.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+import ch.njol.skript.lang.Condition;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+
+@Name("Is Adult")
+@Description("Whether or not an entity is an adult.")
+@Examples("player's target is an adult")
+@Since("INSERT VERSION")
+public class CondIsAdult extends Condition {
```
Comment (FranKusmiruk) [inline src/main/java/ch/njol/skript/conditions/CondIsAdult.java:42]: This is a PropertyCondition of PropertyType.IS

File: src/main/java/ch/njol/skript/conditions/CondIsAdult.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+	
+	static {
+		Skript.registerCondition(CondIsAdult.class, "%entity% (1¦is|2¦is(n't| not)) [a[n]] adult");
```
Comment (FranKusmiruk) [inline src/main/java/ch/njol/skript/conditions/CondIsAdult.java:45]: May as well be worth adding a baby pattern and call it CondAge?

File: src/main/java/ch/njol/skript/conditions/CondIsAdult.java (hunk 1 above)
Comment (DeltaRays) [reply src/main/java/ch/njol/skript/conditions/CondIsAdult.java:42]: PropertyType.IS doesn't exist

File: src/main/java/ch/njol/skript/conditions/CondIsAdult.java (hunk 2 above)
Comment (DeltaRays) [reply src/main/java/ch/njol/skript/conditions/CondIsAdult.java:45]: Wouldn't it be better to make it an expression then?

Comment (DeltaRays) [reply src/main/java/ch/njol/skript/conditions/CondIsAdult.java:45]: %entity's age%

File: src/main/java/ch/njol/skript/conditions/CondIsAdult.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+		entities = (Expression<Entity>) exprs[0];
+		setNegated(parser.mark == 2);
+		return true;
+	}
+	
+	@Override
+	public boolean check(Event e) {
+		if (entities.getAnd())
+			return entities.stream(e).allMatch(entity -> {
+				if (entity instanceof Ageable) return ((Ageable) entity).isAdult();
```
[the remark this answers is not in the document]
Comment (DeltaRays) [reply src/main/java/ch/njol/skript/conditions/CondIsAdult.java:63]: What do you think about just having an expression called %entity's age%?

File: src/main/java/ch/njol/skript/conditions/CondIsAdult.java (hunk 2 above)
Comment (APickledWalrus) [reply src/main/java/ch/njol/skript/conditions/CondIsAdult.java:45]: IMO, I would just keep it as a condition for age over an expression.

File: src/main/java/ch/njol/skript/conditions/CondIsAdult.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+				else if (entity instanceof Zombie) return !((Zombie) entity).isBaby();
+				else return false;
+			}) != isNegated();
+	}
+	
+	@Override
+	public String toString(@Nullable Event e, boolean debug) {
+		return entities.toString(e, debug) + (isNegated() ? " isn't " : " is ") + "an adult";
+	}
+}
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/conditions/CondIsAdult.java:79]: newline at the bottom? 🙂

File: src/main/java/ch/njol/skript/conditions/CondIsAdult.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+@Examples("player's target is an adult")
+@Since("INSERT VERSION")
+public class CondIsAdult extends PropertyCondition<Entity> {
+	static {
+		register(CondIsAdult.class, PropertyType.BE, "adult", "entity");
+	}
+	
+	@Override
+	public boolean check(Entity entity) {
+		if (entity instanceof Ageable) return ((Ageable) entity).isAdult();
```
Comment (TheLimeGlass) [inline src/main/java/ch/njol/skript/conditions/CondIsAdult.java:43]: Please follow Skript's code standards. single condition statements need their section on a separate line from the condition.

File: src/main/java/ch/njol/skript/conditions/CondIsAdult.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+		else if (entity instanceof Zombie) return !((Zombie) entity).isBaby();
+		else return false;
+	}
+	
+	@Override
+	protected String getPropertyName() {
+		return "is adult";
```
Comment (TheLimeGlass) [inline src/main/java/ch/njol/skript/conditions/CondIsAdult.java:50]: The property name is the "adult" that you define in the register method. You don't need to define the word `is` the property condition class handles this.

**Comment by ShaneBeee** [comment]
So I have a few issues with this:
1) You're not checking all possible baby-able entities. Ie: piglins and hoglins. They do not extend from ageable or zombie. In the EntityData class we had to manually add these. 
2) If they were added, the problem now becomes having to do this in the future. For example, when i did the 1.16 update, I had to add the entities to SimpleEntityData, plus I had to add them in for the EntityData class to be able to spawn a baby variant. So the issue comes when having to update future versions of Minecraft, we have to go thru other classes to add the baby/adult stuff there as well. In a case like this, maybe we should add an entity util class somewhere to globablly handle this?
3) This is already possible to do in Skript without this condition.
ie:
```vb
if target entity of player is a baby entity:
if target entity of player is an adult entity:
if target entity of player is a baby zombie:
if target entity of player is an adult sheep:
```
Skript already registers entity data's for these types, and using the compare condition, this can already be done. 
This condition does not add anything new to Skript, nor does it really simplify anything.

**Comment by DeltaRays** [comment]
Okay, then it's not really worth it, I didn't know you could do that

**Pull request closed without merging**